src/libos/src/fs/file_ops/chmod.rs

```diff
@@ -53,6 +53,15 @@ impl FileMode {
     pub fn has_set_gid(&self) -> bool {
         self.contains(FileMode::S_ISGID)
     }
+
+    pub fn to_umask(mut self) -> Self {
+        self.remove(Self::S_ISUID | Self::S_ISGID | Self::S_ISVTX);
```

Comment:
https://man7.org/linux/man-pages/man2/umask.2.html

The doc said that the behavior should be: self & 0777. Please add it as comments to help reviewer to understand the logic.

Reply:
Sure